src/entries/popup/pages/hw/gridplus/walletCredentials.tsx

```diff
@@ -0,0 +1,182 @@
+import { motion } from 'framer-motion';
+import { setup } from 'gridplus-sdk';
+import { FormEvent, useEffect, useState } from 'react';
+
+import { i18n } from '~/core/languages';
+import { useGridPlusClientStore } from '~/core/state/gridplusClient';
+import { LocalStorage } from '~/core/storage';
+import { Box, Button, Text } from '~/design-system';
+import { Input } from '~/design-system/components/Input/Input';
+import { Spinner } from '~/entries/popup/components/Spinner/Spinner';
+import {
+  getStoredGridPlusClient,
+  setStoredGridPlusClient,
+} from '~/entries/popup/handlers/gridplus';
+
+export type WalletCredentialsProps = {
+  appName: string;
+  onAfterSetup?: (result: boolean) => void;
+};
+
+export const WalletCredentials = ({
+  appName,
+  onAfterSetup,
+}: WalletCredentialsProps) => {
+  const setClient = useGridPlusClientStore((state) => state.setClient);
+  const [connecting, setConnecting] = useState(false);
+  const [formData, setFormData] = useState({
+    deviceId: '',
+    password: '',
+  });
+  const formDataFilled =
+    formData.deviceId.length > 0 && formData.password.length > 0;
+  const disabled = !formDataFilled || connecting;
+  const setStoredClient = (storedClient: string | null) => {
+    if (!storedClient) return;
+    setStoredGridPlusClient(storedClient);
+    setClient(storedClient);
+  };
+  const onSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setConnecting(true);
+    try {
+      let result: boolean;
+      if (process.env.IS_TESTING === 'true') {
+        result = true;
+      } else {
+        result = await setup({
+          deviceId: formData.deviceId,
+          password: formData.password,
+          name: appName,
+          getStoredClient: () => useGridPlusClientStore.getState().client,
```

Comment:
what's the diference of `useGridPlusClientStore` and `getStoredGridPlusClient`?

Reply:
Hi @greg-schrammel! Our SDK currently only supports synchronous storages (it's Web Storage API compliant), so in order to upgrade from it, to async Chrome Storage API, we had to utilize a simple Zustand store to make it synchronous.